src/main/java/uk/nhs/digital/nhsconnect/lab/results/model/edifact/LaboratoryInvestigationResult.java

```diff
@@ -11,48 +11,79 @@
 
+    private final String code;
+    private final CodingType codeType;
+    private final String description;
+
+    @SuppressWarnings("checkstyle:MagicNumber")
     public static LaboratoryInvestigationResult fromString(String edifactString) {
-        if (!edifactString.startsWith(KEY_QUALIFIER)) {
+        if (!edifactString.startsWith(KEY)) {
             throw new IllegalArgumentException("Can't create "
-                    + LaboratoryInvestigationResult.class.getSimpleName() + " from " + edifactString);
+                + LaboratoryInvestigationResult.class.getSimpleName() + " from " + edifactString);
         }
 
         final String[] keySplit = Split.byPlus(edifactString);
 
-        final BigDecimal measurementValue = extractMeasurementValue(keySplit);
-        final String measurementValueComparator = extractMeasurementValueComparator(keySplit);
-        final String measurementUnit = extractMeasurementUnit(keySplit);
-        final String deviatingResultIndicator = extractDeviatingResultIndicator(keySplit);
+        final LaboratoryInvestigationResultType resultType =
+            LaboratoryInvestigationResultType.fromCode(keySplit[1]);
+
+        final LaboratoryInvestigationResultBuilder laboratoryInvestigationResultBuilder =
+            LaboratoryInvestigationResult.builder()
+                .resultType(resultType);
 
-        return LaboratoryInvestigationResult.builder()
-            .measurementValue(measurementValue)
-            .measurementValueComparator(StringUtils.isNotBlank(measurementValueComparator)
+        if (resultType.equals(LaboratoryInvestigationResultType.NUMERICAL_VALUE)) {
+            final BigDecimal measurementValue = extractMeasurementValue(keySplit);
+            final String measurementValueComparator = extractMeasurementValueComparator(keySplit);
+            final String measurementUnit = extractMeasurementUnit(keySplit);
+            final String deviatingResultIndicator = extractDeviatingResultIndicator(keySplit);
+
+            return laboratoryInvestigationResultBuilder
+                .measurementValue(measurementValue)
+                .measurementValueComparator(StringUtils.isNotBlank(measurementValueComparator)
                     ? MeasurementValueComparator.fromCode(measurementValueComparator)
                     : null)
-            .measurementUnit(measurementUnit)
-            .deviatingResultIndicator(StringUtils.isNotBlank(deviatingResultIndicator)
+                .measurementUnit(measurementUnit)
+                .deviatingResultIndicator(StringUtils.isNotBlank(deviatingResultIndicator)
                     ? DeviatingResultIndicator.fromCode(deviatingResultIndicator)
                     : null)
+                .build();
+        }
+
+        final String code = Split.byColon(keySplit[2])[2];
+        final String codeType = Split.byColon(keySplit[2])[3];
+        final String description = Split.byColon(keySplit[2])[5];
```

Comment:
Throughout this class, we seem to change between using "magic numbers" as indices or constant values instead, as well as sometimes using methods to extract the values from the segment. Should we change this to be consistent throughout?

Reply:
The reason I left those named constants there is because they're used in private methods (outside of the public method), so at least they still give you some context what they mean